lib/ContractInteract/BrandedToken.js

```diff
@@ -0,0 +1,416 @@
+    if (!(web3 instanceof Web3)) {
+      throw new TypeError("Mandatory Parameter 'web3' is missing or invalid");
+    }
+    if (!Web3.utils.isAddress(address)) {
+      throw new TypeError(
+        `Mandatory Parameter 'address' is missing or invalid: ${address}`,
+      );
+    }
+
+    this.web3 = web3;
+    this.address = address;
+
+    this.contract = Contracts.getBrandedToken(this.web3, this.address);
+
+    if (!this.contract) {
+      throw new TypeError(
+        `Could not load branded token contract for: ${this.address}`,
+      );
+    }
+
+    this.convertToBrandedTokens = this.convertToBrandedTokens.bind(this);
+    this.requestStake = this.requestStake.bind(this);
+    this.requestStakeRawTx = this.requestStakeRawTx.bind(this);
+    this.acceptStakeRequest = this.acceptStakeRequest.bind(this);
+    this.acceptStakeRequestRawTx = this.acceptStakeRequestRawTx.bind(this);
+    this.liftRestriction = this.liftRestriction.bind(this);
+    this.liftRestrictionRawTx = this.liftRestrictionRawTx.bind(this);
+    this.isUnrestricted = this.isUnrestricted.bind(this);
+    this.rejectStakeRequest = this.rejectStakeRequest.bind(this);
+    this.rejectStakeRequestRawTx = this.rejectStakeRequestRawTx.bind(this);
+  }
+
+  /**
+   * Deploys a Branded token contract.
+   *
+   * @dev Conversion parameters provide the conversion rate and its scale.
+   *      For example, if 1 value token is equivalent to 3.5 branded
+   *      tokens (1:3.5), _conversionRate == 35 and
+   *      _conversionRateDecimals == 1.
+   *
+   *      Constructor requires:
+   *          - valueToken address is not zero
+   *          - conversionRate is not zero
+   *          - conversionRateDecimals is not greater than 5
+   *
+   * @param web3 Origin chain web3 object.
+   * @param valueToken The value to which valueToken is set.
+   * @param symbol The value to which tokenSymbol, defined in EIP20Token,
+   *                is set.
+   * @param name The value to which tokenName, defined in EIP20Token,
+   *              is set.
+   * @param decimals The value to which tokenDecimals, defined in EIP20Token,
+   *                  is set.
+   * @param conversionRate The value to which conversionRate is set.
+   * @param conversionRateDecimals The value to which
+   *                                conversionRateDecimals is set.
+   * @param organization Organization contract address.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<BrandedToken>} Promise containing the Branded token
+   *                                  instance that has been deployed.
+   */
+  static async deploy(
+    web3,
+    valueToken,
+    symbol,
+    name,
+    decimals,
+    conversionRate,
+    conversionRateDecimals,
+    organization,
+    txOptions,
+  ) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
```

Comment:
Since the validations for typechecks are repetitive, I think we should create Validation library/util where we can refactor the code. Validation library/util can also be used in openst.js also. @schemar What do you think?

Reply:
That could help 👍